io_ogre/ogre/scene.py

```diff
@@ -414,49 +413,246 @@ def ogre_document(materials):
+                    Report.warnings.append("Unable to create SkyBox: Linked node is not an environment texture. Node type: %s" % linked_node.type)
+                    return None
+            else:
+                # Retrieve the static color if no image is linked
+                background_color = color_input.default_value
+                logger.warning("Unable to create SkyBox: Found background color instead of environment texture")
+                Report.warnings.append("Unable to create SkyBox: Found background color instead of environment texture")
+                return None
+        else:
+            logger.warning("Unable to create SkyBox: No Background node found")
+            Report.warnings.append("Unable to create SkyBox: No Background node found")
+            return None
+    else:
+        logger.warning("Unable to create SkyBox: World nodes are not enabled.")
+        Report.warnings.append("Unable to create SkyBox: World nodes are not enabled.")
+        return None
+
+    skybox_imagepath = linked_node.image.filepath
+    skybox_name = os.path.splitext(linked_node.image.name)[0]
+
+    logger.info("* Generating SkyBox: %s" % skybox_name)
+    logger.info("+ From Image: %s" % skybox_name)
+    logger.info("+ With resolution: %s" % skybox_resolution)
+
+    # Create a collection to render the SkyBox
+    skybox_collection = bpy.data.collections.new(collection_name)
+    bpy.context.scene.collection.children.link(skybox_collection)
+
+    # Create camera to render the SkyBox
+    camera_name = "OgreSkyBox_CAM"
+
+    camera = bpy.data.cameras.new(camera_name)
+    camera_ob = bpy.data.objects.new(camera_name, camera)
+    #bpy.context.scene.objects.link(camera_ob)
+
+    #camera_ob = bpy.data.objects['CameraY']
+    skybox_collection.objects.link(camera_ob)
+
+    scene_camera_orig = scene.camera
+    scene.camera = camera_ob
+
+    # Set SkyBox camera settings
+    camera_ob.data.lens_unit = 'FOV'
+    camera_ob.data.angle = math.radians(90)
+
+    # Depth of Field
+    #camera_ob.data.dof.use_dof = True
+    #camera_ob.data.dof.focus_distance = 10.0
+
+    camera_ob.data.clip_start = 0.1
+    camera_ob.data.clip_end = 1000
+
+    # Backup scene settings
+    scene_res_x_orig = scene.render.resolution_x
+    scene_res_y_orig = scene.render.resolution_y
+    scene_rdr_perc_orig = scene.render.resolution_percentage
+    scene_filepath_orig = scene.render.filepath
+    scene_use_nodes_orig = scene.use_nodes
+    scene_file_format_orig = scene.render.image_settings.file_format
+
+    # Set scene settings for SkyBox rendering
+    scene.render.resolution_x = skybox_resolution
+    scene.render.resolution_y = skybox_resolution
+    scene.render.resolution_percentage = 100
+    scene.use_nodes = True
+    scene.render.image_settings.file_format = 'PNG'
+
+    # Create SkyBox camera orientations
+    front = mathutils.Euler((math.radians(90), 0, 0), 'XYZ')
```

Comment:
these are the orientations used internally by Ogre:
https://github.com/OGRECave/ogre/blob/6bcf1d7d4c510104f794d779e63d330c2c8d5ebb/OgreMain/src/OgreCompositorChain.cpp#L267

the blender orientations might be different due to axis flipping. Maybe we should disable axis flipping when it comes to .scene contents altogether for simplicity.

Reply:
You are right. I'll see how to acomodate the axis swapping with this.

In my opinion the current systems works good, because when you export a .scene what you get is what you see in Blender. Things have the "same" position and orientation, at least visually. And to me and probably most people (I guess) that is the most intuitive output.
Besides this exporter has worked like that for a loong time.

The only issue is that by default lights in Blender point towards the "bottom" or down while in OGRE by default they point towards the "back" of the screen.